Mathlib/Tactic/NormNum/Basic.lean

```diff
@@ -543,3 +543,28 @@ such that `norm_num` successfully recognises both `a` and `b`. -/
   have nc : Q(ℕ) := mkRawNatLit (na.natLit! / nb.natLit!)
   haveI' : Nat.div $na $nb =Q $nc := ⟨⟩
   return .isNat sℕ nc q(isNat_natDiv $pa $pb (.refl $nc))
+
+theorem isNat_dvd_true : {a b : ℕ} → {a' b' : ℕ} →
+    IsNat a a' → IsNat b b' → Nat.mod b' a' = nat_lit 0 → a ∣ b
```

Comment:
What's the heuristic for using `nat_lit` vs `OfNat.ofNat`?

Reply:
Using simpler terms means less work for the kernel, which is going to be normalizing the LHS to a nat literal.